test/server/server_test.cc

```diff
@@ -180,6 +221,8 @@ class ServerInstanceImplTest : public testing::TestWithParam<Network::Address::I
     EXPECT_TRUE(server_->api().fileSystem().fileExists("/dev/null"));
   }
 
+  void flushStats() { server_->flushStats(); }
```

Comment:
I understand why you did this, but instead of reaching into the class under test, manually calling flush, looking at logs, etc. it would be a much better test if you used either one of the existing stat sinks or a simple mock/custom one to test this functionality. Can you work on that? I don't think it's that hard. Normally I wouldn't be asking for this level of testing, but this is pretty important functionality and IMO we should do our best to have a good test.

Reply:
I thought the rest of the code is well tested with existing tests and the new flow needs to be tested  - that is why I went with that approach. Now I have added `CustomStatsSink`. PTAL